src/test/datascience/variableView/variableView.vscode.test.ts

```diff
@@ -100,6 +103,21 @@ suite('DataScience - VariableView', function () {
             { name: 'test2', type: 'str', length: '12', value: ' MYTESTVALUE2' }
         ];
         await waitForVariablesToMatch(expectedVariables, variableView);
+
+        // Verify we don't have any new variables apart from test, test2, os & sys
+        const kernel = kernelProvider.get(cell.notebook.uri)!;
+        const outputs = await kernel.executeHidden('%who_ls');
+        // Sample output is `["test", "test2", "os", "sys"]`
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const vars = ((outputs[0].data as any)['text/plain'] as string)
+            .trim()
```

Comment:
Added a new test to ensure we don't leave any messy variables behind

Reply:
Filed a bug to remove os & sys (stuff that we add when starting kernels & the like)